includes/internal/quiz-submission/answer/repositories/class-aggregate-answer-repository.php

```diff
@@ -47,32 +57,42 @@ class Aggregate_Answer_Repository implements Answer_Repository_Interface {
 	 *
 	 * @internal
 	 *
-	 * @param Comments_Based_Answer_Repository $comments_based_repository Comments based quiz answer repository implementation.
-	 * @param Tables_Based_Answer_Repository   $tables_based_repository  Tables based quiz answer repository implementation.
-	 * @param bool                             $use_tables  The flag if the tables based implementation is available for use.
+	 * @param Comments_Based_Answer_Repository   $comments_based_repository Comments based quiz answer repository implementation.
+	 * @param Tables_Based_Answer_Repository     $tables_based_repository  Tables based quiz answer repository implementation.
+	 * @param Tables_Based_Submission_Repository $tables_based_submission_repository Tables based quiz submission repository.
+	 * @param bool                               $use_tables  The flag if the tables based implementation is available for use.
 	 */
-	public function __construct( Comments_Based_Answer_Repository $comments_based_repository, Tables_Based_Answer_Repository $tables_based_repository, bool $use_tables ) {
-		$this->comments_based_repository = $comments_based_repository;
-		$this->tables_based_repository   = $tables_based_repository;
-		$this->use_tables                = $use_tables;
+	public function __construct(
+		Comments_Based_Answer_Repository $comments_based_repository,
+		Tables_Based_Answer_Repository $tables_based_repository,
+		Tables_Based_Submission_Repository $tables_based_submission_repository,
+		bool $use_tables
+	) {
+		$this->comments_based_repository          = $comments_based_repository;
+		$this->tables_based_repository            = $tables_based_repository;
+		$this->tables_based_submission_repository = $tables_based_submission_repository;
+		$this->use_tables                         = $use_tables;
 	}
 
 	/**
 	 * Create a new answer.
 	 *
 	 * @internal
 	 *
-	 * @param int    $submission_id The submission ID.
-	 * @param int    $question_id   The question ID.
-	 * @param string $value         The answer value.
+	 * @param Submission $submission  The submission.
+	 * @param int        $question_id The question ID.
+	 * @param string     $value       The answer value.
 	 *
 	 * @return Answer The answer model.
 	 */
-	public function create( int $submission_id, int $question_id, string $value ): Answer {
-		$answer = $this->comments_based_repository->create( $submission_id, $question_id, $value );
+	public function create( $submission, int $question_id, string $value ): Answer {
```

Comment:
Should we type hint the `$submission` here as well?

```suggestion
	public function create( Submission $submission, int $question_id, string $value ): Answer {
```